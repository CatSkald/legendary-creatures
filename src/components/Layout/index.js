import "../../styles/global.scss";
import "../../styles/theme-light.scss";
import "../../styles/theme-dark.scss";
import styles from "./index.module.scss";

import React, { useState } from "react";
import Header from "../Header";
import Footer from "../Footer";

const LocaleContext = React.createContext();

const Layout = ({ children, pageContext: { language } }) => {
  const [isDarkColorTheme, switchColorTheme] = useState(false);

  function handleToggleColorTheme() {
    switchColorTheme(!isDarkColorTheme);
  }

  return (
    <LocaleContext.Provider value={{ language }}>
      <div
        className={`body ${styles.Wrapper} ${
          isDarkColorTheme ? "theme-dark" : "theme-light"
        }`}
      >
        <Header
          isDarkColorTheme={isDarkColorTheme}
          handleToggleColorTheme={handleToggleColorTheme}
        />
        <section className={styles.SiteContent} role="main">
          <div className={styles.Container}>{children}</div>
        </section>
        <Footer />
      </div>
    </LocaleContext.Provider>
  );
};

export { Layout, LocaleContext };
