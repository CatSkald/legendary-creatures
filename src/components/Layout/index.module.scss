.layout__container {
  display: flex;
  min-height: 100vh;
  flex-direction: column;
}

.layout {
  display: flex;
  flex: 1;
  width: 100%;
}

.layout__content {
  width: 100%;
  max-width: var(--container-width);
  margin: 0 auto;
  padding: var(--space-sm);

  @media screen and (min-width: 768px /*--screen-medium*/) {
    padding: var(--space);
  }
}
