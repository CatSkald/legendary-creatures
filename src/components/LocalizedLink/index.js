import React from "react";
import { Link } from "gatsby";
import { LocaleContext } from "../Layout";

const LocalizedLink = ({ to, ...props }) => {
  const { language } = React.useContext(LocaleContext);

  const enrichLinkWithLocale = () => {
    // Don't add redundant trailing slash in case of the homepage
    const isHomepage = to === "/";
    return language.path + (isHomepage ? "" : to);
  };

  const path = language.default ? to : enrichLinkWithLocale();

  return <Link {...props} to={path} />;
};

export default LocalizedLink;
