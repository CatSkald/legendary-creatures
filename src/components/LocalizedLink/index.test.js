import React from "react";
import { render } from "@testing-library/react";
import each from "jest-each";

import { LocaleContext } from "../Layout";
import LocalizedLink from "../LocalizedLink";

describe("LocalizedLink", () => {
  each([
    ["/", "en"],
    ["", "ru"],
    ["/uk/", "uk"],
  ]).it("renders correctly for path %s and code %s", (path, code) => {
    const language = { path: path, code: code };
    const { container } = render(
      <LocaleContext.Provider value={{ language }}>
        <LocalizedLink to="my-url" />
      </LocaleContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders correct props", () => {
    const language = { path: "/", code: "en" };
    const { container } = render(
      <LocaleContext.Provider value={{ language }}>
        <LocalizedLink to="my-url" className="my-class" aria-label="test" />
      </LocaleContext.Provider>,
    );
    expect(container).toMatchSnapshot();
  });

  each(["/", "/en/", "en", ""]).it(
    "renders link to homepage correctly",
    (path) => {
      const language = { path: path, code: "en" };
      const { container } = render(
        <LocaleContext.Provider value={{ language }}>
          <LocalizedLink to="/" />
        </LocaleContext.Provider>,
      );
      expect(container).toMatchSnapshot();
    },
  );
});
