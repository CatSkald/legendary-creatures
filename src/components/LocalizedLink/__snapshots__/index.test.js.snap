// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalizedLink renders correct props 1`] = `
<div>
  <a
    aria-label="test"
    class="my-class"
    href="/my-url"
    hreflang="en"
  />
</div>
`;

exports[`LocalizedLink renders correctly for path  and code ru 1`] = `
<div>
  <a
    href="my-url"
    hreflang="ru"
  />
</div>
`;

exports[`LocalizedLink renders correctly for path / and code en 1`] = `
<div>
  <a
    href="/my-url"
    hreflang="en"
  />
</div>
`;

exports[`LocalizedLink renders correctly for path /uk/ and code uk 1`] = `
<div>
  <a
    href="/uk/my-url"
    hreflang="uk"
  />
</div>
`;

exports[`LocalizedLink renders link to homepage correctly 1`] = `
<div>
  <a
    href="/"
    hreflang="en"
  />
</div>
`;

exports[`LocalizedLink renders link to homepage correctly 2`] = `
<div>
  <a
    href="/en/"
    hreflang="en"
  />
</div>
`;

exports[`LocalizedLink renders link to homepage correctly 3`] = `
<div>
  <a
    href="en"
    hreflang="en"
  />
</div>
`;

exports[`LocalizedLink renders link to homepage correctly 4`] = `
<div>
  <a
    href="/"
    hreflang="en"
  />
</div>
`;
